pdf-converter/app/pdf_processor.py

```
@@ -0,0 +1,139 @@
+        
+        texts = completion.choices[0].message.content
+        
+        start = texts.find('[')
+        end = texts.rfind(']')
+        if start != -1 and end != -1 and end > start:
+            texts = texts[start:end+1]
+        
+        try:
+            return json.loads(texts)
+        except json.JSONDecodeError as e:
+            raise ValueError(f"Failed to parse OCR response: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)